src/main/java/ch/njol/skript/expressions/ExprResult.java

```diff
@@ -0,0 +1,131 @@
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionList;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.function.DynamicFunctionReference;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.util.Executable;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Name("Result (Experimental)")
+@Description({"Runs something (like a function) and returns its result.",
+		"If the thing is expected to return multiple values, specify 'results'."})
+@Examples({"set {_function} to the function named \"myFunction\"",
+		"set {_result} to the result of {_function}",
+		"set {_list::*} to the results of {_function}",
+		"set {_result} to the result of {_function} with arguments 13 and true"})
+@Since("INSERT VERSION")
+@Keywords({"run", "result", "execute", "function", "reflection"})
+@SuppressWarnings("NotNullFieldNotInitialized")
+public class ExprResult extends PropertyExpression<Executable<Event, Object>, Object> {
+
+	static {
+		Skript.registerExpression(ExprResult.class, Object.class, ExpressionType.SIMPLE,
+			"[the] result[plural:s] of %executable% [arguments:with arg[ument]s %-objects%]");
+	}
+
+	private Expression<?> arguments;
+	private boolean hasArguments, isPlural;
+	private DynamicFunctionReference.Input input;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(final Expression<?>[] expressions, final int matchedPattern, final Kleenean isDelayed,
+						final SkriptParser.ParseResult result) {
+		if (!this.getParser().hasExperiment(Feature.SCRIPT_REFLECTION))
+			return false;
+		this.setExpr((Expression<? extends Executable<Event, Object>>) expressions[0]);
+		this.hasArguments = result.hasTag("arguments");
+		this.isPlural = result.hasTag("plural");
```

Comment:
Any way we can use the function signature to determine plurality?

Reply:
Sadly no, we don't know the function in the syntax context, we only know about its (maybe) parameters. I suppose we can kind of estimate during runtime based on the signature bootstraps we have seen, but not with any degree of accuracy.
It's annoying, but this was all I could think of.